R/smooth.R

```diff
@@ -53,30 +53,47 @@ smooth_incidence <- function(incidence_data,
 #' @inheritParams inner_module
 #' @param data_points_incl integer. Size of the window used in the LOESS algorithm.
 #' @param degree integer. LOESS degree.
+#' @param initial_Re_estimate_window integer. In order to help with the smoothing, the function extends
+#' the data back in time, padding with values obtained by assuming a constant Re. This parameter represents 
+#' the number of timesteps in the beginning of \code{incidence_input} to take into account when computing
+#' the average initial Re.
 #'
 #' @return module output object. Smoothed incidence.
-.smooth_LOESS <- function(incidence_input, data_points_incl = 21, degree = 1) {
-
+.smooth_LOESS <- function(incidence_input, data_points_incl = 21, degree = 1, initial_Re_estimate_window = 5) {
+  
   .are_valid_argument_values(list(list(incidence_input, "module_input"),
                                   list(data_points_incl, "non_negative_number"),
-                                  list(degree, "non_negative_number"))) #minimal test; needs to be one of {0,1,2}, but stats::loess already throws if it isn't
-
+                                  list(degree, "non_negative_number"), #minimal test; needs to be one of {0,1,2}, but stats::loess already throws if it isn't
+                                  list(initial_Re_estimate_window, "positive_integer")))
+  
   incidence_vector <- .get_values(incidence_input)
-
+  
   n_points <- length(incidence_vector)
   sel_span <- data_points_incl / n_points
-
+  
   n_pad <- round(length(incidence_vector) * sel_span * 0.5)
-
-  c_data <- data.frame(value = c(rep(0, n_pad), incidence_vector),
+  
+  avg_change_rate <- incidence_vector[2:(initial_Re_estimate_window + 1)]/incidence_vector[1:initial_Re_estimate_window]
+  avg_change_rate[!is.finite(avg_change_rate)] <- 1
+  avg_change_rate <- mean(avg_change_rate)
+  
+  previous_day_assumed_cases <- incidence_vector[1]/avg_change_rate
+  values_to_pad_with <- c(previous_day_assumed_cases)
+  
+  for (i in 1:(n_pad-1)){
```

Comment:
In the same line as the previous comment, we can skip this for-loop by using logs.

Reply:
or not even logs but also powers (which are in the end the same)